### PR TITLE
5 add date converter

### PR DESCRIPTION
**Pull request by SirSerje**

### CODE REVIEW

**Review by MaximGordiyenko — DISMISSED** [review]

**Comment by MaximGordiyenko** [comment]
@VadimShy ### can you have a check please agreements on this issue.**
`const twelveTotwenty = function (twelve) {
    for (let i = 0; i < twelve.length; i++) {
        console.log(twelve[i]);
    }
    const hours = twelve[0] + twelve[1];
    const minutes = twelve[3] + twelve[4];
    console.log(Number(hours));
    console.log(Number(minutes));
    const hoursCondition = hours >= 0 && hours <= 12;
    const minutesCondition = minutes >= 0 && minutes < 60;

    if (hoursCondition && minutesCondition) {
        return `${Number(hours) + 12}:${minutes}`;
    }else {
        return "формат до 12 часов до 59 минут";
    }
};
twelveTotwenty("09:31");

const twentyToTwelve = function (twenty) {
    for (let i = 0; i < twenty.length; i++) {
        console.log(twenty[i]);
    }
    const hours = twenty[0] + twenty[1];
    const minutes = twenty[3] + twenty[4];
    console.log(Number(hours));
    console.log(Number(minutes));
    const hoursCondition = hours >= 12 && hours <= 24;
    const minutesCondition = minutes >= 0 && minutes < 60;
    if (hoursCondition && minutesCondition) {
        return `${Number(hours) - 12}:${minutes}`;
    }else {
        return "формат после 12 часов до 59 минут";
    }
};
twentyToTwelve("21:59");`

**Comment by SirSerje** [comment]
@MaximGordiyenko thanks for nice comment, 
@VadimShy don't give up, you can do this ;F)

**Comment by VadimShy** [comment]
> @MaximGordiyenko thanks for nice comment,
> @VadimShy don't give up, you can do this ;F)

I tried so hard and got so far, but in the end ... Test suite failed ))

**Comment by SirSerje** [comment]
@VadimShy If your functions pass more than half of cases, you can remove another ones

**Comment by SirSerje** [comment]
Also, before commit run `yarn lint-fix` and review your code on other syntax/code-style issues.

File: app/utils/dateConverter.js [hunk 1]
```diff
@@ -1,21 +1,33 @@
 const twelveTotwenty = function(twelve) {
     
-  var strToNum = twelve.length > 4 ? +twelve.substr(0, 2) : +twelve.substr(0, 1);
-  var a = strToNum + 12 == 24 ? a = '00' : strToNum + 12; 
-  var b = strToNum > 9 ? twelve.substr(2, 3) : twelve.substr(1, 3);
+  var strToNum = twelve.length > 6 ? +twelve.substr(0, 2) : +twelve.substr(0, 1);
```
Comment (SirSerje) [inline app/utils/dateConverter.js]: use `const` instead of `var`

File: app/utils/dateConverter.js [hunk 2]
```diff
@@ -1,21 +1,33 @@
...
+  var AmCheck = twelve.slice(-2).toLowerCase();
```
Comment (SirSerje) [inline app/utils/dateConverter.js]: we are using `lowerCamelCase` so `amCheck` not `AmCheck`

File: app/utils/dateConverter.js [hunk 3]
```diff
@@ -1,21 +1,33 @@
...
+  var PmCheck = twelve.slice(-2).toLowerCase();
```
Comment (SirSerje) [inline app/utils/dateConverter.js]: we are using `lowerCamelCase` so `amCheck` not `AmCheck`

File: app/utils/dateConverter.js [hunk 4]
```diff
@@ -1,21 +1,33 @@
...
+  var a = strToNum + 12 == 24 ? a = '00' : AmCheck == 'am' ? strToNum = '0' + strToNum : strToNum + 12;
```
Comment (SirSerje) [inline app/utils/dateConverter.js]: I don't understand meaning of `a`. Please give to variable more clean name

File: app/utils/dateConverter.js [hunk 5]
```diff
@@ -1,21 +1,33 @@
 const twelveTotwenty = function(twelve) {
     
-  var strToNum = twelve.length > 4 ? +twelve.substr(0, 2) : +twelve.substr(0, 1);
-  var a = strToNum + 12 == 24 ? a = '00' : strToNum + 12; 
-  var b = strToNum > 9 ? twelve.substr(2, 3) : twelve.substr(1, 3);
+  var strToNum = twelve.length > 6 ? +twelve.substr(0, 2) : +twelve.substr(0, 1);
+  var AmCheck = twelve.slice(-2).toLowerCase();
+  var PmCheck = twelve.slice(-2).toLowerCase();
+  var a = strToNum + 12 == 24 ? a = '00' : AmCheck == 'am' ? strToNum = '0' + strToNum : strToNum + 12; 
+  var b = strToNum > 9 ? twelve.substr(2, 3) : (strToNum <= 9 && twelve.length > 4) ? twelve.substr(2, 3) : twelve.substr(1, 3);
```
Comment (SirSerje) [inline app/utils/dateConverter.js]: I don't understand meaning of `b`. Please give to variable more clean name

File: app/utils/dateConverter.js [hunk 6]
```diff
@@ -1,21 +1,33 @@
...
   var twenty = a + b; 
-  
+
+    if (strToNum > 24 || twelve.length == 5) {
```
Comment (SirSerje) [inline app/utils/dateConverter.js]: Record like `if (strToNum > 24 || twelve.length == 5) twenty = null;` will be shorter. Depends on you

File: app/utils/dateConverter.js [hunk 7]
```diff
@@ -1,21 +1,33 @@
...
     
-  var strToNum = +twenty.substr(0, 2);
-  var a = (strToNum % 12) || 12;
-  var amPm = strToNum < 12 ? "AM" : "PM";
-  var twelve = a + twenty.substr(2, 3) + amPm;
-  
-  return twelve;
-}
+    var strToNum = twenty.length > 4 ? +twenty.substr(0, 2) : +twenty.substr(0, 1);
+    var a = (strToNum % 12) || 12;
```
Comment (SirSerje) [inline app/utils/dateConverter.js]: Give proper name instead of `a`. Can be `let` or `const`

File: app/utils/dateConverter.js [hunk 8]
```diff
@@ -1,21 +1,33 @@
...
+    var b = (strToNum > 9) ? twenty.substr(2, 3) : (strToNum <= 9 & twenty.length > 4) ? twenty.substr(2, 3) : twenty.substr(1, 3);
```
Comment (SirSerje) [inline app/utils/dateConverter.js]: Same as above

File: app/utils/dateConverter.js [hunk 9]
```diff
@@ -1,21 +1,33 @@
...
+    var amPm = strToNum < 12 ? " AM" : strToNum == 24 ? " AM" : " PM";
+    var twelve = a + b + amPm;
+
+    if (strToNum < 0 || strToNum > 24 || twenty.length > 5) {
+      twelve = null;
```
Comment (SirSerje) [inline app/utils/dateConverter.js]: 👍 nice

**Review by SirSerje — COMMENTED** [review]
Good approach, you've successfully done all test cases. Now you can fix linting and do fixes after review

**Comment by SirSerje** [comment]
Lint rules already fixed on dev, review completed, merge id `develop`

**Pull request merged**